ledger/src/blockstore.rs

```diff
@@ -7166,51 +7141,50 @@ pub mod tests {
             .unwrap();
 
         let slot_meta = blockstore.meta(0).unwrap().unwrap();
-        let shred5 = shreds[5].clone();
-
-        // Ensure that an empty shred (one with no data) would get inserted. Such shreds
-        // may be used as signals (broadcast does so to indicate a slot was interrupted)
-        // Reuse shred5's header values to avoid a false negative result
-        let empty_shred = Shred::new_from_data(
-            shred5.slot(),
-            shred5.index(),
-            {
-                let parent_offset = shred5.slot() - shred5.parent().unwrap();
-                parent_offset as u16
-            },
-            &[], // data
-            ShredFlags::LAST_SHRED_IN_SLOT,
-            0, // reference_tick
-            shred5.version(),
-            shred5.fec_set_index(),
+
+        // make a "blank" FEC set that would normally be used to terminate the block
+        let terminator = shredder
+            .entries_to_shreds(
+                &Keypair::new(),
+                &[],
+                true,
+                // chained_merkle_root
+                Some(Hash::new_from_array(rand::thread_rng().gen())),
+                6, // next_shred_index,
+                6, // next_code_index
+                true,
+                &rsc,
+                &mut ProcessShredsStats::default(),
+            )
+            .0;
+
+        assert!(
+            blockstore.should_insert_data_shred(
+                &terminator[0],
+                &slot_meta,
+                &HashMap::new(),
+                max_root,
+                None,
+                ShredSource::Repaired,
+                &mut Vec::new(),
+            ),
+            "Inserting a shred from terminator block should succeed"
         );
-        assert!(blockstore.should_insert_data_shred(
```

Comment:
this tried to check if we can re-insert slot termination shred a second time when it is coming over repair. Sadly we do not have an easy way to set appropriate repair nonce for merkle shreds. Not sure if adding such API is justified